app/controllers/courses/applications_open_controller.rb

```diff
@@ -22,23 +21,35 @@ def continue
     def errors; end
 
     def actual_params
-      params.require(:course).permit(
-        :applications_open_from,
-        :day,
-        :month,
-        :year,
-      )
+      params.require(:course)
+        .except(
+          :qualification,
+          :maths,
+          :english,
+          :science,
+          :funding_type,
+          :level,
+          :is_send,
+          :study_mode,
+          :age_range_in_years,
+        )
+        .permit(
+          :applications_open_from,
+          :day,
+          :month,
+          :year,
+        )
     end
 
-    def course_params
+    def build_course_params
```

Comment:
It could be the lack of sleep, but I had to bend my mind backwards to understand what this was all about. Might be worth a short comment to help explain it's about the dates optionally being specified as year/month/day.

Reply:
Yeah i can see that - i did wonder how to name this, comment will probably help a good amount though